LinqToDBPostGisNetTopologySuite.Tests/LinearReferencingTests.cs

```diff
@@ -0,0 +1,318 @@
+                .Value(g => g.Geometry, () => GeometryInput.STGeomFromEWKT(geomText))
+                .Insert();
+
+                var result = db.TestGeometries
+                .Where(g => g.Id == 1)
+                .Select(g => g.Geometry.STLocateBetween(1.5, 3.0, 0.0))
+                .Single() as NTSG.GeometryCollection;
+
+                Assert.IsNotNull(result);
+                Assert.AreEqual(2, result.NumGeometries);
+
+                var geom1 = result.Geometries[0];
+                var geom2 = result.Geometries[1];
+
+                Assert.IsInstanceOf<NTSG.LineString>(geom1);
+                Assert.IsInstanceOf<NTSG.Point>(geom2);
+
+                var resLineString = geom1 as NTSG.LineString;
+                var resPoint = geom2 as NTSG.Point;
+
+                Assert.AreEqual(1.0, resLineString.Coordinates[0].X, 1.0E-8);
+                Assert.AreEqual(2.0, resLineString.Coordinates[0].Y, 1.0E-8);
+                Assert.AreEqual(3.0, resLineString.Coordinates[0].M, 1.0E-8);
+
+                Assert.AreEqual(3.0, resLineString.Coordinates[1].X, 1.0E-8);
+                Assert.AreEqual(4.0, resLineString.Coordinates[1].Y, 1.0E-8);
+                Assert.AreEqual(2.0, resLineString.Coordinates[1].M, 1.0E-8);
+
+                Assert.AreEqual(9.0, resLineString.Coordinates[2].X, 1.0E-8);
+                Assert.AreEqual(4.0, resLineString.Coordinates[2].Y, 1.0E-8);
+                Assert.AreEqual(3.0, resLineString.Coordinates[2].M, 1.0E-8);
+
+                Assert.AreEqual(1.0, resPoint.X, 1.0E-8);
+                Assert.AreEqual(2.0, resPoint.Y, 1.0E-8);
+                Assert.AreEqual(3.0, resPoint.M, 1.0E-8);
+            }
+        }
+
+        [Test]
+        public void TESTSTLocateBetweenElevations()
+        {
+            var geomText = "LINESTRING(1 2 3, 4 5 6)";
+            using (var db = new PostGisTestDataConnection(TestDatabaseConnectionString))
+            {
+                db.TestGeometries
+                .Value(g => g.Id, 1)
+                .Value(g => g.Geometry, () => GeometryInput.STGeomFromEWKT(geomText))
+                .Insert();
+
+                var result = db.TestGeometries
+                .Where(g => g.Id == 1)
+                .Select(g => g.Geometry.STLocateBetweenElevations(2.0, 4.0))
+                .Single() as NTSG.MultiLineString;
+
+                Assert.IsNotNull(result);
+                Assert.AreEqual(1, result.NumGeometries);
+
+                Assert.AreEqual(1.0, result.Coordinates[0].X, 1.0E-8);
+                Assert.AreEqual(2.0, result.Coordinates[0].Y, 1.0E-8);
+                Assert.AreEqual(3.0, result.Coordinates[0].Z, 1.0E-8);
+
+                Assert.AreEqual(2.0, result.Coordinates[1].X, 1.0E-8);
+                Assert.AreEqual(3.0, result.Coordinates[1].Y, 1.0E-8);
+                Assert.AreEqual(4.0, result.Coordinates[1].Z, 1.0E-8);
+            }
+        }
+
+        [Test]
+        public void TESTSTInterpolatePoint()
```

Comment:
Naming convention for functions is PascalCase:
`TestSTInterpolatePoint`

Reply:
Will search some vs/code extensions' help.   😂 Actually the code has been through many times review by myself.....